### PR TITLE
(case 20005) Create HFMSerializer interface for model loaders

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/19302/improve-model-loading-tool-chain-project-Phase-1
Sub tickets:
https://highfidelity.fogbugz.com/f/cases/20005/Create-HFMSerializer-interface-for-model-loaders

As part of Part 1 of the Model Improvement Project, this PR streamlines the inputs/outputs to the model loading classes as seen by ModelCache, though a new universal interface called HFMSerializer (typedefd from hfm::Serializer). FBXReader and similar classes are re-named to fit this interface, for example FBXReader becomes FBXSerializer, and so on. A new file in the shared library called HifiTypes is also created, which HFMSerializer uses. This PR is a follow-up to https://github.com/highfidelity/hifi/pull/14348, which created the HFM library.

## TEST PLAN
- Run these asset browser baking tests:
   - https://highfidelity.testrail.net/index.php?/cases/view/17647
   - https://highfidelity.testrail.net/index.php?/cases/view/17645
- Run these automatic tests:
  - https://github.com/highfidelity/hifi_tests/tree/master/tests/content/entity/model/modelReaders
- In this PR, verify collision models work as expected for "Basic" and "Good" collision types
  - Note: This test is different from the one in https://github.com/highfidelity/hifi/pull/14156
  - For both "Basic" and "Good" collision types:
    - Use this model url: https://gist.githubusercontent.com/sabrina-shanman/457b40f2a66b517b0dc41196bf6ab96e/raw/bf5405a23cefe3662e30d14edf4d3386a406e128/wamConsole_hull_2_grouptest.obj
    - Create the model. In a domain with edit rights, open Create Tools. Under Create tab, select Model, paste the model url into the Model URL field, then select either "Basic" or "Good" from the Automatic Collisions dropdown. Click "Add"
    - Resize the model if needed so it is near the scale of the avatar. The model should look like a grey, chair-shaped object. Then, make the avatar attempt to stand on the seat area of the model.
  - For "Basic" collision, the avatar should fail to stand on the seat area as there is an invisible sloped barrier in the way
  - For "Good" collision, the avatar should be able to stand on the seat area.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15132/HighFidelity-Beta-PR14394-e08bc72.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15132/HighFidelity-Beta-PR14394-e08bc72.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15132/HighFidelity-Beta-Interface-PR14394-e08bc72.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15146/HighFidelity-Beta-PR14394-2d30902.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15146/HighFidelity-Beta-PR14394-2d30902.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15146/HighFidelity-Beta-Interface-PR14394-2d30902.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15146/HighFidelity-Beta-PR14394-2d30902.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15146/HighFidelity-Beta-Interface-PR14394-2d30902.exe

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15320/HighFidelity-Beta-PR14394-be51137.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15320/HighFidelity-Beta-PR14394-be51137.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15320/HighFidelity-Beta-Interface-PR14394-be51137.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15320/HighFidelity-Beta-PR14394-be51137.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15320/HighFidelity-Beta-Interface-PR14394-be51137.exe

**Comment by samcake** [comment]
Please merge this asap
more useful testing in integration

**Comment by qaAdamSmith82** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15376/HighFidelity-Beta-PR14394-be51137.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15376/HighFidelity-Beta-PR14394-be51137.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15376/HighFidelity-Beta-Interface-PR14394-be51137.exe

File: libraries/animation/src/AnimationCache.h [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #include <QtScript/QScriptValue>
 
 #include <DependencyManager.h>
-#include <FBXReader.h>
+#include <FBXSerializer.h>
```
[the remark this answers is not in the document]
Comment (sabrina-shanman) [reply libraries/animation/src/AnimationCache.h]: No, I will move that to the source file in the next PR.

File: libraries/fbx/src/FBXSerializer.h [hunk 2]
```diff
@@ -147,9 +142,9 @@ class FBXReader {
 
     void consolidateHFMMaterials(const QVariantHash& mapping);
 
-    bool _loadLightmaps = true;
-    float _lightmapOffset = 0.0f;
-    float _lightmapLevel;
+    bool _loadLightmaps { true };
```
[the remark this answers is not in the document]
Comment (sabrina-shanman) [reply libraries/fbx/src/FBXSerializer.h]: They are used in FBXSerializer::consolidateHFMMaterials

**Comment by sabrina-shanman** [comment]
Build this please

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/15396/HighFidelity-Beta-PR14394-be51137.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15396/HighFidelity-Beta-PR14394-be51137.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/15396/HighFidelity-Beta-Interface-PR14394-be51137.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15396/HighFidelity-Beta-PR14394-be51137.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/15396/HighFidelity-Beta-Interface-PR14394-be51137.exe

**Pull request merged**